### DISCUSSION TITLE
Sample Application using Spring AI's Function Calling + RAG

### DISCUSSION THREAD

**Discussion by pankajtandon** [spring-projects/spring-ai#1082] [Show and tell]
Hi all,
Here is an app I created that combines both Function Calling + RAG in such a way that current and personal questions can be answered efficiently by OpenAI's LLM. The RAG approach reduces the the function metadata that is sent to the LLM.

https://github.com/pankajtandon/vacationplanner

I've implemented this with OpenAI's and Mistral's models but have found that `gpt-4-0125-preview` performs much better.

Hope this helps someone!

**Comment by ogbozoyan** [comment]
Thanks for your work
